src/condition.cpp

```diff
@@ -134,6 +134,18 @@ void conditional_t<T>::set_has_activity( bool is_npc )
     };
 }
 
+template<class T>
+void conditional_t<T>::set_is_riding( bool is_npc )
+{
+    condition = [is_npc]( const T & d ) {
+        Character *actor = dynamic_cast<Character *>( d.alpha );
```

Comment:
```suggestion
        return ( is_npc_ ? d.alpha : d.beta )->is_mounted();
```
Less (repeated) code and no need for casting to `Character`.